modules/base/src/processors/volumesource.cpp

```diff
@@ -98,6 +98,11 @@ VolumeSource::VolumeSource(InviwoApplication* app, const std::string& file)
     addProperties(file_, reader_, reload_, information_, basis_, volumeSequence_);
     volumeSequence_.setVisible(false);
 
+    // ensure that the file name is always serialized by setting the default to ""
+    if (!file.empty()) {
```

Comment:
Does this make sense? Should be
```suggestion
    if (file.empty()) {
```

Reply:
no, in case the file name is empty, the default is an empty string "" as well. We need to override the default in case `filePath` is set. This is due to inherent mechanism of property constructors.
If the file name is set for the constructor, it will not be serialized unless changed. See #1390

Reply:
No need for the if statement then

Reply:
What about the overhead of `Property::setStateAsDefault()`? That is not necessary if no file name was set.
https://github.com/inviwo/inviwo/blob/903f8b9e1253c4a7b8acb79c2bd234661bccb7e3/include/inviwo/core/properties/property.h#L469